sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnInvokers.java

```diff
@@ -0,0 +1,460 @@
+      throw new RuntimeException("Unable to find UserCodeException.wrap", e);
+    }
+
+    return new StackManipulation() {
+      @Override
+      public boolean isValid() {
+        return tryBody.isValid();
+      }
+
+      @Override
+      public Size apply(MethodVisitor mv, Implementation.Context implementationContext) {
+        Label tryBlockStart = new Label();
+        Label tryBlockEnd = new Label();
+        Label catchBlockStart = new Label();
+        Label catchBlockEnd = new Label();
+
+        String throwableName = new TypeDescription.ForLoadedType(Throwable.class).getInternalName();
+        mv.visitTryCatchBlock(tryBlockStart, tryBlockEnd, catchBlockStart, throwableName);
+
+        // The try block attempts to perform the expected operations, then jumps to success
+        mv.visitLabel(tryBlockStart);
+        Size trySize = tryBody.apply(mv, implementationContext);
+        mv.visitJumpInsn(Opcodes.GOTO, catchBlockEnd);
+        mv.visitLabel(tryBlockEnd);
+
+        // The handler wraps the exception, and then throws.
+        mv.visitLabel(catchBlockStart);
+        // Add the exception to the frame
+        mv.visitFrame(
+            Opcodes.F_SAME1,
+            // No local variables
+            0,
+            new Object[] {},
+            // 1 stack element (the throwable)
+            1,
+            new Object[] {throwableName});
+
+        Size catchSize =
+            new Compound(MethodInvocation.invoke(createUserCodeException), Throw.INSTANCE)
+                .apply(mv, implementationContext);
+
+        mv.visitLabel(catchBlockEnd);
+        // The frame contents after the try/catch block is the same
+        // as it was before.
+        mv.visitFrame(
+            Opcodes.F_SAME,
+            // No local variables
+            0,
+            new Object[] {},
+            // No new stack variables
+            0,
+            new Object[] {});
+
+        return new Size(
+            trySize.getSizeImpact(),
+            Math.max(trySize.getMaximalSize(), catchSize.getMaximalSize()));
+      }
+    };
+  }
+
+  /**
+   * A constructor {@link Implementation} for a {@link DoFnInvoker class}. Produces the byte code
+   * for a constructor that takes a single argument and assigns it to the delegate field.
+   */
+  private static final class InvokerConstructor implements Implementation {
+    @Override
+    public InstrumentedType prepare(InstrumentedType instrumentedType) {
+      return instrumentedType;
+    }
+
+    @Override
+    public ByteCodeAppender appender(final Target implementationTarget) {
+      return new ByteCodeAppender() {
+        @Override
+        public Size apply(
+            MethodVisitor methodVisitor,
+            Context implementationContext,
+            MethodDescription instrumentedMethod) {
+          StackManipulation.Size size =
+              new StackManipulation.Compound(
+                      // Load the this reference
```

Comment:
I used https://github.com/google/google-java-format and I'd prefer not to tweak the indentation manually on top of that.